### PR TITLE
Transform ElementFinderInstigator into an interface `TypedInstigator`

### PR DESCRIPTION
**Pull request by juankysoriano**
Continue with the work that @blundell started on #35 

This is a PR to make `ElementFinderInstigator` a `Streamer` so we can achieve flexibility when attempting to parse without any callbacks. This is specially important when working with complex XML files as we might require working with multiple `ElementFinder`s instead of just one (as `ElementFinderInstigator` was doing. 

The name `Streamer` comes from the idea of having a component which *streams* the xml. Such operation is likely to be performed for element finding or xml parsing.

### CODE REVIEW

**Comment by xrigau** [comment]
:+1:

File: sexp/src/main/java/com/novoda/sexp/Instigator.java [hunk 1]
```diff
@@ -13,8 +13,9 @@
     RootTag getRootTag();
 
     /**
-     * This is where you create the 'tree' of
+     * This is where you traverse the xml 'tree' by using
```
Comment (blundell) [inline sexp/src/main/java/com/novoda/sexp/Instigator.java]: this is wrong tho isn't it? you don't traverse anything, the xml is fed to you

File: sexp/src/main/java/com/novoda/sexp/Streamer.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+     * <li>{@link com.novoda.sexp.marshaller.AttributeMarshaller AttributeMarshaller}</li>
+     * <li>{@link com.novoda.sexp.marshaller.BodyMarshaller BodyMarshaller}</li>
+     * <li>{@link com.novoda.sexp.parser.Parser Parser}</li>
+     * <li>{@link com.novoda.sexp.parser.ParseWatcher ParseWatcher}</li>
+     * <li>{@link com.novoda.sexp.parser.ListParseWatcher ListParseWatcher}</li>
+     * </ul>
+     *
+     * @param rootElement the root element of your XML file
+     */
+    void stream(RootElement rootElement);
```
Comment (blundell) [inline sexp/src/main/java/com/novoda/sexp/Streamer.java]: i don't think this terminology makes sense in terms of pull parsers, but I'd need to work on it to know any better , so ship it!

**Comment by blundell** [comment]
:x: you didn't update the README https://github.com/novoda/simple-easy-xml-parser/blob/master/README.md , need to tick "check strings and comments" in your refactoring menu :-)  or search for `ElementFinderInstigator`

**Comment by juankysoriano** [comment]
@blundell , You are completely right with your comments. 

I am going to open a PR in relation to the inclusion of `ListAttributeMarshaller` that we did as part of the benchmark testing. I will take the chance to fix the comments and double check where else could that have happened.

File: sexp/src/main/java/com/novoda/sexp/Streamer.java (hunk 2 above)
Comment (juankysoriano) [inline sexp/src/main/java/com/novoda/sexp/Streamer.java]: We based our bet for this name after reading the following http://www.extreme.indiana.edu/xgws/xsoap/xpp/

> Xml Pull Parser (in short XPP) is a streaming pull XML parser and should be used when there is a need to process quickly and efficiently all input elements (for example in SOAP processors). Pull parsing is just one level up from tokenizing XML (and that is actually how XPP2 default implementation is done).

**Comment by juankysoriano** [comment]
I have updated the `README.md` to reflect this changes.

File: sexp/src/main/java/com/novoda/sexp/Streamer.java (hunk 2 above)
Comment (blundell) [inline sexp/src/main/java/com/novoda/sexp/Streamer.java]: ok fair enough :+1:

**Comment by blundell** [comment]
:+1:

**Pull request merged**